pkg/statser/statser_internal.go

```diff
@@ -34,7 +34,7 @@ type InternalStatser struct {
 
 // NewInternalStatser creates a new Statser which sends metrics to the
 // supplied InternalHandler.
-func NewInternalStatser(bufferSize int, tags gostatsd.Tags, namespace, hostname string, handler InternalHandler) Statser {
+func NewInternalStatser(bufferSize int, tags gostatsd.Tags, namespace, hostname string, handler InternalHandler) *InternalStatser {
```

Comment:
I'm not liking the change on the return type, specifically as it will be made configurable later on.